momento-sdk/src/main/java/momento/sdk/UserHeaderInterceptor.java

```diff
@@ -35,6 +40,7 @@ public void start(Listener<RespT> listener, Metadata metadata) {
         metadata.put(AUTH_HEADER_KEY, tokenValue);
         if (!isUserAgentSent) {
           metadata.put(SDK_AGENT_KEY, sdkVersion);
+          metadata.put(RUNTIME_VERSION_KEY, runtimeVer);
```

Comment:
just wondering, it looks like we are setting these in 2 different places? What is the reasoning behind that?

Reply:
I noticed that too. But this is the pattern we were using: Clients use the "metadataWithCache" from ScsClientBase class to create headers, whereas the stub managers use the UseHeaderInterceptor class to create the headers.

Reply:
I think you only need to add the headers once. It might make more sense to use the metadataWithCache style that you added and get rid of this interceptor, because `isUserAgentSent` in this class is static, so it will only add the headers to the first momento client that executes a call. If you see `java:cache:1.0.0`, you'll never see `java:store:1.0.0`.